File: sdk/go/profiles.go
package wlteopenapi

import "context"

type ProfilesService struct {
	client *Client
}

func (s *ProfilesService) List(ctx context.Context) (DeviceProfileList, error) {
	var result DeviceProfileList
	err := s.client.request(ctx, "GET", "/wlte/v1/device-profiles", nil, nil, nil, &result)
	return result, err
}

File: sdk/typescript/src/profiles.ts
import type { WlteClient } from './client.js'
import type { ApiEnvelope, DeviceProfileList } from './types.js'

export class ProfilesApi {
  constructor(private readonly client: WlteClient) {}

  async list(): Promise<DeviceProfileList> {
    const response = await this.client.request<ApiEnvelope<DeviceProfileList>>('/wlte/v1/device-profiles')
    return response.data
  }
}

